zcnbridge/http/rest.go

```diff
@@ -8,72 +8,115 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"sync"
 
+	"go.uber.org/zap"
+
+	"github.com/0chain/gosdk/core/logger"
 	"github.com/0chain/gosdk/core/util"
 	"github.com/0chain/gosdk/zcnbridge/errors"
-	"github.com/0chain/gosdk/zcnbridge/log"
 	"github.com/0chain/gosdk/zcncore"
 )
 
 const (
 	// SCRestAPIPrefix represents base URL path to execute smart contract rest points.
-	SCRestAPIPrefix      = "v1/screst/"
-	SmartContractAddress = `6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7`
-	RestPrefix           = SCRestAPIPrefix + SmartContractAddress
-	GetAuthorizersPath   = "/getAuthorizerNodes"
+	SCRestAPIPrefix        = "v1/screst/"
+	SmartContractAddress   = `6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7`
+	RestPrefix             = SCRestAPIPrefix + SmartContractAddress
+	PathGetAuthorizerNodes = "/getAuthorizerNodes"
+	PathGetGlobalConfig    = "/getGlobalConfig"
+	PathGetAuthorizer      = "/getAuthorizer"
 )
 
+type Params map[string]string
+
+var Logger logger.Logger
+var defaultLogLevel = logger.DEBUG
+
+func init() {
+	Logger.Init(defaultLogLevel, "0chain-zcnbridge-sdk")
+}
+
 // MakeSCRestAPICall calls smart contract with provided address
 // and makes retryable request to smart contract resource with provided relative path using params.
-func MakeSCRestAPICall(relativePath string, params map[string]string) ([]byte, error) {
+func MakeSCRestAPICall(opCode int, relativePath string, params Params, cb zcncore.GetInfoCallback) {
 	var (
 		resMaxCounterBody []byte
 		hashMaxCounter    int
+		msg               string
 		hashCounters      = make(map[string]int)
 		sharders          = extractSharders()
-		lastErrMsg        string
 	)
 
+	type queryResult struct {
+		hash string
+		body []byte
+	}
+
+	results := make(chan *queryResult)
```

Comment:
Use buffered channel, the  size should be the same as the len(sharders), otherwise only the first response could goto the channel, all others will get stuck. And because they are stuck, the whole function could get stuck on the wg.Wait().

Reply:
done